# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../linux26/20-kernel-cc.patch
# Copyright (C) 2004 - 2008 The T2 SDE Project
# Copyright (C) 1998 - 2003 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

This patch is specific to the T2 SDE build system to not apply all the
agressive compiler options, but only less, kernel specific ones.
Patch initially for ROCK Linux by Clifford Wolf <god@clifford.at>.

--- linux/Makefile
+++ linux/Makefile
@@ -27,7 +27,9 @@
 
 AS		= $(CROSS_COMPILE)as
 LD		= $(CROSS_COMPILE)ld
-CC		= $(CROSS_COMPILE)gcc
+KCC		= $(CROSS_COMPILE)kcc
+GCC		= $(CROSS_COMPILE)gcc
+CC		= $(if $(shell sh -c 'type -p $(KCC)'),$(KCC),$(GCC))
 CPP		= $(CC) -E
 AR		= $(CROSS_COMPILE)ar
 NM		= $(CROSS_COMPILE)nm
